pandas/io/parquet.py

```diff
@@ -104,19 +104,26 @@ def write(
             from_pandas_kwargs["preserve_index"] = index
 
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
+
+        fs = get_fs_for_path(path)
```

Comment:
This feels like it should be done at a higher level in the base class and passed through.

Reply:
Call it `filesystem`.

Reply:
`FastParquetImpl` doesn't require this as it uses `get_file_and_filesystem` to obtain the filesystem in the `read` method.

`fs` is used in the `read` method for `PyArrowImpl`. To be consistent with it, `fs` is a better choice.